transit_odp/timetables/csv.py

```diff
@@ -302,6 +387,83 @@ def add_seasonal_status(df: pd.DataFrame) -> pd.DataFrame:
     return annotated_df
 
 
+def add_staleness_metrics(df: pd.DataFrame, today: datetime.date) -> pd.DataFrame:
+    today = np.datetime64(today)
+    df["last_modified_date"] = df["modification_datetime"].dt.date
+    df["effective_last_modified_date"] = df["last_modified_date"]
+
+    df["effective_seasonal_start"] = df["seasonal_start"] - pd.Timedelta(days=42)
+
+    df["effective_stale_date_from_end_date"] = df[
+        "operating_period_end_date"
+    ] - pd.Timedelta(days=42)
+    defer_one_year = (
+        lambda d: d if pd.isna(d) else datetime.date(d.year + 1, d.month, d.day)
+    )
+    df["effective_stale_date_from_last_modified"] = df[
+        "effective_last_modified_date"
+    ].apply(defer_one_year)
+    df["effective_stale_date_from_otc_effective"] = df["effective_date"] - pd.Timedelta(
+        days=42
+    )
+
+    df["last_modified_lt_effective_stale_date_otc"] = (
+        df["last_modified_date"] < df["effective_stale_date_from_otc_effective"]
+    )
+
+    staleness_end_date = (
+        pd.notna(df["effective_stale_date_from_end_date"])
+        & (
+            df["effective_stale_date_from_end_date"]
+            < df["effective_stale_date_from_last_modified"]
+        )
+        & (df["effective_stale_date_from_end_date"] <= today)
+        & (df["last_modified_date"] >= df["effective_date"])
+    )
+    staleness_12_months = (
+        (
+            pd.isna(df["effective_stale_date_from_end_date"])
+            | (
+                df["effective_stale_date_from_last_modified"]
+                < df["effective_stale_date_from_end_date"]
+            )
+        )
+        & (df["effective_stale_date_from_last_modified"] <= today)
+        & (df["last_modified_date"] >= df["effective_date"])
+    )
+    staleness_otc = (df["last_modified_date"] < df["effective_date"]) & (
+        df["effective_stale_date_from_otc_effective"] < today
```

Comment:
Should be <= on line 435, `df["effective_stale_date_from_otc_effective"] <= today`?

Reply:
Correctamundo!